packages/react-components/react-progress/src/components/Progress/useProgressState.tsx

```diff
@@ -0,0 +1,35 @@
+import type { ProgressState, ProgressProps } from './Progress.types';
+import { progressCssVars } from './useProgressStyles';
+
+// if the percentComplete is near 0, don't animate it.
+// This prevents animations on reset to 0 scenarios
+const ZERO_THRESHOLD = 0.01;
+
+export const useProgressState_unstable = (state: ProgressState, props: ProgressProps) => {
+  const { percentComplete = -1 } = props;
+
+  const determinate = percentComplete > -1 ? true : false;
+
+  const valuePercent = determinate ? Math.min(100, Math.max(0, percentComplete)) : undefined;
```

Comment:
If you take a percent complete and it is not between 0 and 100, it might be better to throw an error.